algosdk/future/transaction.py

```diff
@@ -215,8 +215,12 @@ def undictify(d):
             args.update(PaymentTxn._undictify(d))
             txn = PaymentTxn(**args)
         elif txn_type == constants.keyreg_txn:
-            args.update(KeyregTxn._undictify(d))
```

Comment:
We remove `KeyregTxn._undictify` method as this is the only place where this method is called, and undictify is dependent on the type of KeyregTxn. For example, in the offline case, there are no additional arguments to add.